Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs

```diff
@@ -464,7 +469,12 @@ public void UpdatePlaybackAction(PlaybackInfo playInfo)
             sql_query += "AND UserId not IN (select UserId from UserList) ";
             sql_query += "GROUP BY UserId, date ORDER BY UserId, date ASC";
 
-            endDate = endDate.AddHours(-timezoneOffset);
+            // Convert the decimal to minutes
+            // 0.5*60 = 30
+            int minOffset = (int)(( timezoneOffset < 0.0 ? (-timezoneOffset)%1 : timezoneOffset%1 )*60);
+            // Turn to negative in case timezoneOffset is negative
+            minOffset = ( timezoneOffset < 0.0 ? -minOffset : minOffset );
+            endDate = endDate.AddHours((int)-timezoneOffset).AddMinutes(-minOffset);
```

Comment:
since this code seems to be repeated 6, would it not call for a function to then call 6 times instead?

Reply:
Oof, I'm sorry. I should have lol

I just wanted the plugin to work instead of thinking.
Honestly, there's probably an even better way to do it. I'll try when I get some time today. Thank you!